submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/validators/SubmissionProjectValidator.kt

```diff
@@ -0,0 +1,24 @@
+package ac.uk.ebi.biostd.submission.validators
+
+import ac.uk.ebi.biostd.submission.exceptions.InvalidProjectException
+import ac.uk.ebi.biostd.submission.exceptions.MissingProjectAccessTagException
+import ebi.ac.uk.model.ExtendedSubmission
+import ebi.ac.uk.model.extensions.attachTo
+import ebi.ac.uk.persistence.PersistenceContext
+
+class SubmissionProjectValidator : SubmissionValidator {
+    override fun validate(submission: ExtendedSubmission, context: PersistenceContext) {
+        submission.attachTo?.let {
+            validateProject(it, context)
+            validateProjectAccessTag(it, submission, context)
+        }
+    }
+
+    private fun validateProject(project: String, context: PersistenceContext) =
+        context.getSubmission(project) ?: throw InvalidProjectException(project)
+
+    private fun validateProjectAccessTag(project: String, submission: ExtendedSubmission, context: PersistenceContext) =
```

Comment:
what is the purpose of this validation? the idea AFAIK is that the system will create access tag for the project. Users should not use access tags.

Reply:
This validation is intended for projects that were manually created and the access tag might not exist.